cpp/daal/src/algorithms/dtrees/forest/regression/df_regression_train_dense_default_impl.i

```diff
@@ -380,54 +411,74 @@ bool OrderedRespHelper<algorithmFPType, cpu>::findBestSplitOrderedFeature(const
                                                                           const ImpurityData & curImpurity, TSplitData & split,
                                                                           double minWeightLeaf) const
 {
-    algorithmFPType xi = this->_aResponse[aIdx[0]].val;
     ImpurityData left;
-    left.var  = 0;
-    left.mean = xi;
-
     ImpurityData right;
-    calcPrevImpurity<algorithmFPType, cpu>(curImpurity.var * algorithmFPType(n), curImpurity.mean, right.var, right.mean, xi, n);
-
+    algorithmFPType xi           = this->_aResponse[aIdx[0]].val;
+    left.var                     = 0;
+    left.mean                    = xi;
+    algorithmFPType nTotalWeight = n;
+    IndexType iBest              = -1;
+    algorithmFPType vBest;
+    auto aResponse = this->_aResponse.get();
+    auto aWeight   = this->_aWeight.get();
+    if (this->_weight)
+    {
+        nTotalWeight = 0.;
+        PRAGMA_VECTOR_ALWAYS
+        for (size_t i = 0; i < n; ++i)
+        {
+            nTotalWeight += aWeight[aIdx[i]].val;
+        }
```

Comment:
Do we really need to re-calculate it every time?

Reply:
No, it was redundant, fixed